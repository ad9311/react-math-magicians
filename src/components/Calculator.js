import React from 'react';
// import calculate from '../logic/calculate';
import './calculator.css';
import Key from './Key';
import manageInput from '../logic/manageInput';
// import PropTypes from 'prop-types';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.input = { total: '', next: '', oparation: '' };
    this.state = { total: '0' };
  }

  handleKeyDown = (keyPressed) => {
    this.manageInput = manageInput(keyPressed, this.input);
    this.manageInput.next();
  }

  render() {
    const click = this.handleKeyDown;
    const { total } = this.state;

    return (
      <div className="keypad">
        <div className="io-container">
          <output className="d-block text-right pr-1">{total}</output>
        </div>
        <div className="row">
          <Key value="AC" handleKeyDown={click} />
          <Key value="+/-" handleKeyDown={click} />
          <Key value="%" handleKeyDown={click} />
          <Key orange value="รท" handleKeyDown={click} />
        </div>
        <div className="row">
          <Key value="7" handleKeyDown={click} />
          <Key value="8" handleKeyDown={click} />
          <Key value="9" handleKeyDown={click} />
          <Key orange value="x" handleKeyDown={click} />
        </div>
        <div className="row">
          <Key value="4" handleKeyDown={click} />
          <Key value="5" handleKeyDown={click} />
          <Key value="6" handleKeyDown={click} />
          <Key orange value="-" handleKeyDown={click} />
        </div>
        <div className="row">
          <Key value="1" handleKeyDown={click} />
          <Key value="2" handleKeyDown={click} />
          <Key value="3" handleKeyDown={click} />
          <Key orange value="+" handleKeyDown={click} />
        </div>
        <div className="row">
          <div className="row">
            <Key value="0" handleKeyDown={click} />
          </div>
          <div className="row">
            <Key value="." handleKeyDown={click} />
            <Key orange value="=" handleKeyDown={click} />
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
