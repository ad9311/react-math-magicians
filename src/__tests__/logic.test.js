import React from 'react';
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import Calculator from '../components/Calculator.js';


describe('It calculates mathematical operations', () => {
  test('5 times 3', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByText('15')).toMatchSnapshot();
  });

  test('15 + 5', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByText('20')).toMatchSnapshot();
  });
});
