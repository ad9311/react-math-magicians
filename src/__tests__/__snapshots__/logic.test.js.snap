// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It calculates mathematical operations 5 times 3 1`] = `
<output
  class="d-block text-right pr-1"
>
  15
</output>
`;

exports[`It calculates mathematical operations 15 + 5 1`] = `
<output
  class="d-block text-right pr-1"
>
  20
</output>
`;
